engine/Assets/Scripts/Utilities/ColorManager/ColorManager.cs

```diff
@@ -0,0 +1,232 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using Synthesis.PreferenceManager;
+using Synthesis.Util;
+using SynthesisAPI.EventBus;
+using UI.Dynamic.Modals.Configuring.ThemeEditor;
+using UnityEngine;
+
+namespace Utilities.ColorManager {
+    public static class ColorManager {
+        public const string SELECTED_THEME_PREF = "color/selected_theme";
+        public const string DEFAULT_THEME       = "Default";
+
+        public class OnThemeChanged : IEvent {}
+
+        private static readonly Color UNASSIGNED_COLOR = new(200, 255, 0, 255);
+
+        private static string THEMES_FOLDER_PATH {
+            get {
+                string dir = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) +
+                             Path.AltDirectorySeparatorChar + "Autodesk" + Path.AltDirectorySeparatorChar +
+                             "Synthesis" + Path.AltDirectorySeparatorChar + "Themes";
```

Comment:
To make this more readable I think we should save the separation character to a variable.

```cs
char altSep = Path.AltDirectorySeparatorChar;
```

Other than that looks good.